app/src/androidTest/java/ch/hikemate/app/ui/map/MapScreenTest.kt

```diff
@@ -351,4 +353,15 @@ class MapScreenTest : TestCase() {
         .onNodeWithTag(MapScreen.TEST_TAG_LOCATION_PERMISSION_ALERT)
         .assertIsNotDisplayed()
   }
+
+  @Test
+  fun mapLoadsFromMapState() {
+    // Uses Mockk since Mockito can not mock/spy final classes
+    val spyHikesViewModel = spyk(hikesViewModel)
+    setUpMap(hikesViewModel = spyHikesViewModel)
+
+    composeTestRule.waitForIdle()
+
+    io.mockk.verify { spyHikesViewModel.getMapState() }
+  }
```

Comment:
Question: Could you explain what does `spyk` do exactly? What's the difference with `mock` ?

Also, Why have you added this test ?

Reply:
The reason I spy here is because if I don't I can not call `verify` on the `hikesViewModel`, since no one keeps track of how often it was called. By using spy, I can keep track of how often the `.getMapState()` was called, but I don't need to go through the effort of mocking the `HikesViewModel`, since it just uses its original functionality.

Reply:
I added the test to ensure that the MapScreen actually calls and makes use of the newly added `MapState`. I noticed that I had not tested this before. Imo its a useful test to ensure that the functionality I implemented works continuously.

Reply:
Thanks for the explanation. More clearer to me now.